topo/node/lemming/lemming.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package lemming contains a node implementation for a lemming device.
+package lemming
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/openconfig/kne/topo/node"
+
+	tpb "github.com/openconfig/kne/proto/topo"
+)
+
+func New(nodeImpl *node.Impl) (node.Node, error) {
+	if nodeImpl == nil {
+		return nil, fmt.Errorf("nodeImpl cannot be nil")
+	}
+	if nodeImpl.Proto == nil {
+		return nil, fmt.Errorf("nodeImpl.Proto cannot be nil")
+	}
+	defaults(nodeImpl.Proto)
+	n := &Node{
+		Impl: nodeImpl,
+	}
+	return n, nil
+}
+
+type Node struct {
+	*node.Impl
+}
+
+func (n *Node) ResetCfg(ctx context.Context) error {
+	return nil
+}
+
+func defaults(pb *tpb.Node) *tpb.Node {
+	if pb.Config == nil {
+		pb.Config = &tpb.Config{}
+	}
+	if pb.Config.Image == "" {
+		pb.Config.Image = "lemming:latest"
+	}
+	if len(pb.GetConfig().GetCommand()) == 0 {
+		pb.Config.Command = []string{"/lemming/lemming"}
+	}
+	if len(pb.GetConfig().GetArgs()) == 0 {
+		pb.Config.Args = []string{"--alsologtostderr"}
+	}
+	if pb.Config.EntryCommand == "" {
+		pb.Config.EntryCommand = fmt.Sprintf("kubectl exec -it %s -- /bin/bash", pb.Name)
+	}
+	if pb.Services == nil {
+		pb.Services = map[uint32]*tpb.Service{
```

Comment:
add ssh?

Reply:
ssh isn't enabled of the lemming container, leaving it off for now